src/runtime_src/xdp/profile/plugin/aie_trace/aie_trace_plugin.cpp

```diff
@@ -1468,22 +1468,247 @@ bool AieTracePlugin::configureStartIteration(xaiefal::XAieMod& core)
+        boost::replace_all(metrics[i][j], "}", "");
+      }
+      std::vector<std::string> minTile;
+      boost::split(minTile, metrics[i][0], boost::is_any_of(","));
+      uint32_t minCol = std::stoi(minTile[0]);
+      uint32_t minRow = std::stoi(minTile[1]);
+
+      std::vector<std::string> maxTile;
+      boost::split(maxTile, metrics[i][1], boost::is_any_of(","));
+      uint32_t maxCol = std::stoi(maxTile[0]);
+      uint32_t maxRow = std::stoi(maxTile[1]);
+
+      for (uint32_t col = minCol; col <= maxCol; ++col) {
+        for (uint32_t row = minRow; row <= maxRow; ++row) {
+          xrt_core::edge::aie::tile_type tile;
+          tile.col = col;
+          tile.row = row;
+//            tiles.push_back(tile);
+          mConfigMetrics[tile] = metrics[i][2];
+//           mConfigMetrics[moduleIdx][tile] = metrics[i][2];
+          // check if same tiles are recognized
+        }
+      }
+    } // Pass 2
+
+    // Pass 3 : process only single tile metric setting 
+    for (size_t i = 0; i < metricsSettings.size(); ++i) {
+
+      if (2 != metrics[i].size()) {
+        continue;
+      }
+      if (0 == metrics[i][0].compare("all")) {
+        continue;
+      }
+      boost::replace_all(metrics[i][0], "{", "");
+      boost::replace_all(metrics[i][0], "}", "");
+
+      std::vector<std::string> tilePos;
+      boost::split(tilePos, metrics[i][0], boost::is_any_of(","));
+
+      xrt_core::edge::aie::tile_type tile;
+      tile.col = std::stoi(tilePos[0]);
+      tile.row = std::stoi(tilePos[1]);
+//        tiles.push_back(tile);
+      mConfigMetrics[tile] = metrics[i][1];
+//       mConfigMetrics[moduleIdx][tile] = metrics[i][1];
+    } // Pass 3 
+
+
+    // check validity and remove "off" tiles
+    std::vector<tile_type> offTiles;
+
+    for (auto &tileMetric : mConfigMetrics) {
+
+      // save list of "off" tiles
+      if (tileMetric.second.empty() || 0 == tileMetric.second.compare("off")) {
+        offTiles.push_back(tileMetric.first);
+        continue;
+      }
+
+      // Ensure requested metric set is supported (if not, use default)
+      if (metricSets.find(tileMetric.second) == metricSets.end()) {
+        std::string defaultSet = "functions";
+        std::stringstream msg;
+        msg << "Unable to find AIE trace metric set " << tileMetric.second 
+            << ". Using default of " << defaultSet << "."
+            << " As new AIE_trace_settings section is given, old style configurations, if any, are ignored.";
+        xrt_core::message::send(severity_level::warning, "XRT", msg.str());
+        tileMetric.second = defaultSet;
+      }
+    }
+
+    // remove all the "off" tiles
+    for (auto &t : offTiles) {
+      mConfigMetrics.erase(t);
+    }
+
+    // If requested, turn on debug fal messages
+    if (xrt_core::config::get_verbosity() >= static_cast<uint32_t>(severity_level::debug))
+      xaiefal::Logger::get().setLogLevel(xaiefal::LogLevel::DEBUG);
+  }
```

Comment:
I think this series of sequential loops can be greatly simplified into nested loops. Let's address that in a future PR.